lib/asn1/gen.c

```diff
@@ -1903,10 +1925,13 @@ generate_type_header (const Symbol *s)
      * member fields are not OPTIONAL/DEFAULTed.
      */
     generate_subtypes_header(s);
-    fprintf(headerfile, "/*\n");
-    fprintf(headerfile, "%s ::= ", s->name);
-    define_asn1 (0, s->type);
-    fprintf(headerfile, "\n*/\n\n");
+    if (!s->emitted_asn1) {
```

Comment:
For every type we emit it's ASN.1 definition in a C comment above it, but we recurse just above (`generate_subtypes_header(s);`) and we were emitting the same commentary over and over.

Reply:
ack